syntax = "proto3";

package user;

service UserService {
    rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
    rpc GetUser (GetUserRequest) returns (GetUserResponse);
    rpc UpdateUser (UpdateUserRequest) returns (UpdateUserResponse);
}

message CreateUserRequest {
    string username = 1;
    string email = 2;
}

message CreateUserResponse {
    string user_id = 1;
    string message = 2;
}

message GetUserRequest {
    string user_id = 1;
}

message GetUserResponse {
    string user_id = 1;
    string username = 2;
    string email = 3;
}

message UpdateUserRequest {
    string user_id = 1;
    string username = 2;
    string email = 3;
}

message UpdateUserResponse {
    string message = 1;
}
